cmd/shim/main.go

```diff
@@ -322,3 +347,87 @@ func toggleNesting(ctx context.Context) ([]string, error) {
 	}
 	return []string{}, nil
 }
+
+type SecretScrubWriter struct {
```

Comment:
nit: Consider moving this to a separate file, `main.go` is starting to get too big

Reply:
Since there was only 1 file, I didn't want to break it down, but it's good for me too.

Reply:
done